docs/contributing/building.rst

```diff
@@ -135,6 +135,63 @@ or source it into your shell,
 
    . $BUILD_DIR/python_env.sh
 
+Resolving Common Build Issues
+=============================
+
+If you have having issues from a clean build directory (i.e., empty directory),
+
+1. Make sure you have also checked out any git submodules: ``git submodule
+   update --recursive --init``
+
+2. If you are using Conda, make sure that you have activated your environment
+   for both the ``cmake`` and ``make`` steps: ``conda activate katana-dev``
+
+If you were previously successful building but now you are seeing ``cmake`` or
+unexpected build errors after updating your source directory,
+
+1. Make sure you have also checked out any git submodules: ``git submodule
+   update --recursive --init``
+
+2. Check if there were any system build environment changes since the last time
+   you successfully built. To update your environment, run
+   ``scripts/setup_dev_ubuntu.sh``.
+
+   This requires root privileges, if you don't have root, it is likely that
+   your system administrator has already updated your build environment.
+
+3. Check if there were any build environment changes since the last time you
+   successfully built.
+
+   For Conda, run ``conda deactivate ; conda env update --name katana-dev
+   --file $SRC_DIR/conda_recipe/environment.yml``. If you have submodules, you
+   will have to run the previous command for the
+   ``conda_recipe/environment.yml`` in each submodule.
+
+   For Conan, run ``conan install $SRC_DIR/config --build=missing``. If you
+   have submodules, you only have to run this command for the main source
+   directory.
+
+4. Remove your cached build variables to pick up on any build environment
+   changes (system or otherwise): ``rm ${BUILD_DIR}/CMakeCache.txt``
+
+5. Run your ``cmake`` command.
+
+   If you are using Conda, make sure you have activated your environment before
+   running ``cmake``.
+
+6. Run ``make``
+
+Careful readers may notice that the above sequence of commands is equivalent to
+creating a new build directory and configuring from scratch. As gain
```

Comment:
This isn't true. The problem is that the instructions above leave around the object files which may actually prevent a future build since they may be linked against an older version of a library. I think cmake in theory tries to avoid this, but I don't think it does very well especially if the library files didn't change name.

I would recommend adding a step above to completely delete your build directory (and clear your ccache) if just reconfiguring doesn't work.

Reply:
- Added a `make clean` and `ccache -C` step.
- Changed the wording around equivalence